.changes/unreleased/Features-20230604-025956.yaml

```
@@ -0,0 +1,7 @@
+kind: Features
+body: Revamp debug, add --connection flag, and drop --config-dir flag. Prepare for
+  future refactors/interface changes.
```

After https://github.com/dbt-labs/dbt-core/pull/7741/commits/92fcffbd54fa16a1b8c61514f2895de53a3a97ec, the changelog entry can be shortened:
```suggestion
body: Revamp debug, add --connection flag. Prepare for future refactors/interface changes.
```